lib/spack/spack/spec.py

```diff
@@ -3137,12 +3358,16 @@ def _constrain_dependencies(self, other):
         for name in self.common_dependencies(other):
             changed |= self[name].constrain(other[name], deps=False)
             if name in self._dependencies:
-                changed |= self._dependencies[name].update_deptypes(
-                    other._dependencies[name].deptypes)
+                # TODO: revisit the lines below, they may need to change
```

Comment:
Elsewhere it is mentioned that multiple dependencies with the same name was not supported for the old concretizer (and this method is only used for concretization) - so I think it would be appropriate to add a warning similar to the one you added to `_concretize_helper`.

Reply:
Done in 2d4673b